src/promptflow/promptflow/_sdk/_service/entry.py

```diff
@@ -51,24 +51,26 @@ def add_show_status_action(subparsers):
     show_status_parser.set_defaults(action="show-status")
 
 
+@monitor_operation(activity_name="pfs.start", activity_type=ActivityType.PUBLICAPI)
 def start_service(args):
     port = args.port
     app, _ = create_app()
-    if port and is_port_in_use(port) and not args.force:
-        app.logger.warning(f"Service port {port} is used.")
-        raise UserErrorException(f"Service port {port} is used.")
-    if not port:
-        port = get_port_from_config(create_if_not_exists=True)
-    else:
-        dump_port_to_config(port)
 
-    if is_port_in_use(port):
-        if args.force:
-            app.logger.warning(f"Force restart the service on the port {port}.")
-            kill_exist_service(port)
-        else:
-            app.logger.warning(f"Service port {port} is used.")
-            raise UserErrorException(f"Service port {port} is used.")
+    def validate_port(port, force_start):
+        if is_port_in_use(port):
+            if force_start:
+                app.logger.warning(f"Force restart the service on the port {port}.")
+                kill_exist_service(port)
+            else:
+                app.logger.warning(f"Service port {port} is used.")
+                raise UserErrorException(f"Service port {port} is used.")
+
+    if port:
+        dump_port_to_config(port)
+        validate_port(port, args.force)
+    else:
+        port = get_port_from_config(create_if_not_exists=True)
+        validate_port(port, args.force)
     # Set host to localhost, only allow request from localhost.
     app.logger.info(f"Start Prompt Flow Service on http://localhost:{port}, version: {get_promptflow_sdk_version()}")
     waitress.serve(app, host="127.0.0.1", port=port)
```

Comment:
This is a blocking api, so our monitor operation never ends? how we know the latency then ?

Reply:
what' the latency here mean?

Reply:
The time used to start the service

Reply:
update pfs start telemetry behavior according to discussion